src/pages/javascript/jquery-widgets/dialog.md

```diff
@@ -0,0 +1,248 @@
+
+**Type**: Boolean
+
+**Default value**: `false`
+
+### `defaultDialogClass`
+
+Class that is added to the drop-down, when it gets initialized.
+
+**Type**: String
+
+**Default value**: `mage-dropdown-dialog`
+
+Class that is added/removed on drop-down content when it gets opened/closed.
+
+**Type**: String
+
+**Default value**: `null`
+
+### `parentClass`
+
+Class that is added/removed for the dropdown parent, when the drop-down gets opened/closed.
+
+**Type**: String
+
+**Default value**: `null`
+
+### `triggerClass`
+
+Class that is added/removed on the trigger element when the drop-down gets opened/closed.
+
+**Type**: String
+
+**Default value**: `null`
+
+### `triggerEvent`
+
+Name of the event that triggers the `open()` function for drop-down.
+
+**Type**: String
+
+**Default value**: `click`
+
+### `triggerTarget`
+
+Element that triggers the drop-down.
+
+**Type**:
+
+-  String
+-  jQuery object
+
+**Default value**: `null`
+
+### `timeout`
+
+The number of milliseconds until the dropdown is closed after mouse pointer moves out.
+
+**Type**: Number
+
+**Default value**: 500
+
+## Methods
+
+The customized dropdownDialog widget has default [jQuery UI Dialog widget] methods, though some of them are customized.
+
+Customized public methods:
+
+-  [open()](#open)
+-  [close()](#close)
+
+### `open()`
+
+Beside default functionality, this method calls the `_mouseLeave()` and `_mouseEnter()` functions, adds classes on trigger and drop-down parent, and binds the close on mouse click outside drop-down to the `<body>` element.
+
+Invoke the open method:
+
+```javascript
+$("#element").dropdownDialog("open");
+```
+
+### `close()`
+
+Beside default functionality, this method removes the classes from trigger and drop-down parent.Also clears the timeout if the latter exists.
+
+Invoke the close method:
+
+```javascript
+$("#element").dropdownDialog("close");
+```
+
+## Code sample
+
+The following example shows how to initialize the dropdown dialog widget and pass options during the initialization.
+This example uses some CSS classes that are used for minicart.
+
+```html
+<div data-block="dropdown" class="minicart-wrapper">
+    <button type="button" class="action" data-trigger="trigger">
+        <span data-bind="i18n: 'Click Here'"></span>
+    </button>
+</div>
+<div class="block block-minicart"
+     data-mage-init='{
+        "dropdownDialog": {
+            "appendTo": "[data-block=dropdown]",
+            "triggerTarget":"[data-trigger=trigger]",
+            "timeout": 2000,
+            "closeOnMouseLeave": false,
+            "closeOnEscape": true,
+            "autoOpen": true,
+            "triggerClass": "active",
+            "parentClass": "active",
+            "buttons": []
+        }
+     }'>
+    <div id="minicart-content-wrapper">
+        Here is our content
+    </div>
+</div>
+```
+
+The following example shows a PHTML file using the script and custom CSS classes that have no defined properties:
+
+```php
```

Comment:
this is outside the scope of this pr, but it seems like every php code gate in this repo (and every other devsite repo) does not use syntax highlighting for `php` code gates. I'll add a post about it in the migration channel.